src/services/redis.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from collections.abc import Generator
+
+from redis.asyncio import Redis as AsyncRedis
+
+from utils.settings import (
+    REDIS_DB_NUMBER,
+    REDIS_HOST,
+    REDIS_PASSWORD,
+    REDIS_PORT,
+    REDIS_SSL_ENABLED,
+)
+from utils.singleton_meta import SingletonMeta
+
+
+class Redis(metaclass=SingletonMeta):
+    """
+    Manages a singleton connection to the Redis database.
+
+    This class establishes a connection to the Redis database using the provided
+    credentials and configuration. It ensures that only one instance of the connection
+    exists at any time by utilizing the SingletonMeta metaclass.
+    """
+
+    connection: AsyncRedis | None = None
+
+    def set_connection(self, connection: AsyncRedis) -> None:
+        """
+        sets the database connection for the service.
+
+        args:
+            connection (AsyncRedis): the database connection object.
+        """
+        self.connection = connection
+
+
+def get_redis_connection() -> Generator[Redis, None, None]:
```

Comment:
What is the purpose of `Generator` here?

Reply:
Oh, now that you ask: It is no longer needed. I restructured the code but overlooked that this would make the generators obsolete. The original intention to close connections in a finally block. But since that is not done in any other place where we open connections, I adapted to that.